src/main/java/teammates/ui/webapi/QueryLogsAction.java

```diff
@@ -76,10 +98,42 @@ public JsonResult execute() {
                 .build();
         try {
             QueryLogsResults queryResults = logsProcessor.queryLogs(queryLogsParams);
+            removeSensitiveFields(queryResults);
             GeneralLogsData generalLogsData = new GeneralLogsData(queryResults);
             return new JsonResult(generalLogsData);
         } catch (LogServiceException e) {
             return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
     }
+
+    private void removeSensitiveFields(QueryLogsResults queryResults) {
+        if (userInfo.isAdmin) {
+            return;
+        }
+
+        for (GeneralLogEntry logEntry : queryResults.getLogEntries()) {
+            if (logEntry.getDetails() != null) {
+                Map<String, Object> details = logEntry.getDetails();
+                // Always remove requestParams, requestHeaders and userInfo for non-admin maintainers
+                details.remove("requestParams");
+                details.remove("requestHeaders");
+                details.remove("userInfo");
+                // Keep log message of event logs and remove log message for other logs for non-admin maintainers
+                if (!details.containsKey("event")) {
+                    details.remove("message");
+                }
+                // Remove email details in email sent event log, and remove student email in feedback session audit
+                // event log for non-admin maintainers
+                if (details.containsKey("event") && details.get("event").equals(LogEvent.EMAIL_SENT.toString())) {
```

Comment:
This can be shortened to just:
```java
LogEvent.EMAIL_SENT.toString().equals(details.get("event"))
```
similarly for the next one.

Reply:
Ok. It is updated.